onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2260,19 +2260,107 @@ def aten_embedding_backward(
     raise NotImplementedError()
 
 
+@torch_op("aten::embedding_bag", trace_only=True)
 def aten_embedding_bag(
-    weight: TensorType,
-    indices: TensorType,
-    offsets: TensorType,
-    scale_grad_by_freq: bool = False,
-    mode: int = 0,
-    sparse: bool = False,
-    per_sample_weights: Optional[TensorType] = None,
+    weight: TFloat,
+    indices: INT64,
+    offsets: INT64 = None,  # Could be None accotding to the doc, go 2d branch
+    scale_grad_by_freq: bool = False,  # pylint: disable=unused-argument
+    mode: int = 0,  # [0,1,2] indicate ["mean", "sum", "max"]
+    sparse: bool = False,  # pylint: disable=unused-argument
+    per_sample_weights: Optional[TFloat] = None,
     include_last_offset: bool = False,
-) -> tuple[TensorType, TensorType, TensorType, TensorType]:
+) -> TFloat:
     """embedding_bag(Tensor weight, Tensor indices, Tensor offsets, bool scale_grad_by_freq=False, int mode=0, bool sparse=False, Tensor? per_sample_weights=None, bool include_last_offset=False) -> (Tensor, Tensor, Tensor, Tensor)"""
 
-    raise NotImplementedError()
+    if per_sample_weights is None:
+        per_sample_weights = op.Expand(op.Constant(value_floats=[1.0]), op.Shape(indices))
+        per_sample_weights = op.CastLike(per_sample_weights, weight)
+
+    if len(indices.shape) == 1:  # 1d
+        return _aten_embedding_bag_1d_onnx(
+            weight, indices, offsets, mode, per_sample_weights, include_last_offset
+        )
+    else:  # 2d
+        # assert(len(indices.shape) == 2)
+        # assert(indices.shape == per_sample_weights.shape)
+        return _aten_embedding_bag_2d_onnx(weight, indices, mode, per_sample_weights)
+
+
+@torch_op("aten::embedding_bag", private=True)
+def _aten_embedding_bag_1d_onnx(
+    weight: TFloat,
+    indices: INT64,
+    offsets: INT64,
+    mode: int,
+    per_sample_weights: TFloat,
+    include_last_offset: bool,
+) -> TFloat:
+    neg_1 = op.Constant(value_ints=[-1])
+    # Get weight out according to indices,
+    # e.g. indices=[0,1,2,4] means get weight[0],weight[1],weight[2],weight[4]
+    new_weight = op.Gather(weight, indices)
+    new_weight = op.Mul(new_weight, op.Unsqueeze(per_sample_weights, axes=1))
+    dim_1_size = op.Reshape(op.Gather(op.Shape(weight), 1), neg_1)
+    # Parse offsets=[0,1,3,3,4]
+    # When include_last_offset=False, means: [0:1],[1:3],[3:3],[3:4],[4:end]
+    # When include_last_offset=True, means: [0:1],[1:3],[3:3],[3:4]
+    len_tensor = op.Reshape(op.Size(offsets), neg_1)
+    if include_last_offset == True:  # pylint: disable=singleton-comparison
+        len_tensor = len_tensor - 1
+    else:
+        offsets = op.Concat(offsets, op.Shape(indices), axis=0)  # Replace end with number
+
+    # The element in sequence must be FLOAT32 dtype due to ORT bug
```

Comment:
Should we file a bug to ORT and link it here?

Reply:
Already have this bug in ORT.